### PR TITLE
[RHACS] Added instructions for enabling RHCOS scanning

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://issues.redhat.com/browse/ROX-12542

- Cherrypick into `rhacs-docs-4.0`

Preview: 
- Scanning RHCOS node hosts: https://59886--docspreview.netlify.app/openshift-acs/latest/operating/manage-vulnerabilities/scan-rhcos-node-host.html#rhcos-enable-node-scan_scan-rhcos-node-host
- Upgrading using the roxctl CLI: https://59886--docspreview.netlify.app/openshift-acs/latest/upgrading/upgrade-roxctl.html#rhcos-enable-node-scan_upgrade-roxctl

### CODE REVIEW

**Comment by gaurav-nelson** [comment]
Thank you @ocpdocs-previewbot 😉

File: modules/rhcos-enable-node-scan.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
...X_CSV_EXPORT","value":"false"},{"name":"ROX_DECLARATIVE_CONFIGURATION","value":"false"},{"name":"ROX_INTEGRATIONS_AS_CONFIG","value":"false"},{"name":"ROX_NETPOL_FIELDS","value":"true"},{"name":"ROX_NETWORK_DETECTION_BASELINE_SIMULATION","value":"true"},{"name":"ROX_NETWORK_GRAPH_PATTERNFLY","value":"true"},{"name":"ROX_NODE_SCANNING_CACHE_TIME","value":"3h36m"},{"name":"ROX_NODE_SCANNING_INITIAL_BACKOFF","value":"30s"},{"name":"ROX_NODE_SCANNING_MAX_BACKOFF","value":"5m"},{"name":"ROX_PROCESSES_LISTENING_ON_PORT","value":"false"},{"name":"ROX_QUAY_ROBOT_ACCOUNTS","value":"true"},{"name":"ROX_ROXCTL_NETPOL_GENERATE","value":"true"},{"name":"ROX_SOURCED_AUTOGENERATED_INTEGRATIONS","value":"false"},{"name":"ROX_SYSLOG_EXTRA_FIELDS","value":"true"},{"name":"ROX_SYSTEM_HEALTH_PF","value":"false"},{"name":"ROX_VULN_MGMT_WORKLOAD_CVES","value":"false"}],"image":"quay.io/stackrox-io/scanner-slim:4.0.0","imagePullPolicy":"IfNotPresent","name":"node-inventory","ports":[{"containerPort":8444,"name":"grpc","protocol":"TCP"}],"volumeMounts":[{"mountPath":"/host","name":"host-root-ro","readOnly":true},{"mountPath":"/tmp/","name":"tmp-volume"},{"mountPath":"/cache","name":"cache-volume"}]}]}}}}'
```
Comment (vikin91) [inline modules/rhcos-enable-node-scan.adoc:39]: The image `quay.io/stackrox-io/scanner-slim:4.0.0` must be parametrized (its value is okay for opensource or dev installations). The default value should be rather: `registry.redhat.io/advanced-cluster-security/rhacs-scanner-rhel8:4.0.0`

**Review by vikin91 — CHANGES_REQUESTED** [review]
The text is okay, but we need to update the image registry to take the downstream image (see the comment) instead of the upstream one.

Another thing is that this feature should be enabled by default and I though this procedure should be a part of the upgrade steps in roxctl, here:

Screenshot 2023-05-11 at 09 32 13 

I think it is okay having it in both places: "enabling the feature" + "upgrading over roxctl". I would say that the second location is more important as users will look there more frequently.

**Comment by vikin91** [comment]
@Maddosaurus could you give this a look as well? The more reviewers the better ;)

File: operating/manage-vulnerabilities/scan-rhcos-node-host.adoc [hunk 2]
```diff
@@ -12,18 +12,20 @@ With {rh-rhacs-first}, you can scan RHCOS nodes for vulnerabilities and detect p
 
 {product-title-short} scans RHCOS RPMs installed on the node host, as part of the RHCOS installation, for any known vulnerabilities.
 
-First, {product-title-short} analyzes and detects RHCOS components, and then it matches vulnerabilities for identified components using Red Hat RHEL and OpenShift 4.X Open Vulnerability and Assessment Language (OVAL) v2 security data streams.
+First, {product-title-short} analyzes and detects RHCOS components, and then it matches vulnerabilities for identified components by using Red Hat RHEL and OpenShift 4.X Open Vulnerability and Assessment Language (OVAL) v2 security data streams.
 
 [NOTE]
 ====
+* You must manually enable the RHCOS node scanning features.
```
Comment (Maddosaurus) [inline operating/manage-vulnerabilities/scan-rhcos-node-host.adoc:19]: So, technically this is only true for manifest-based installations.
Installations managed by Helm and the Operator will get this feature automatically if OpenShift is detected as a platform.

Comment (gaurav-nelson) [reply operating/manage-vulnerabilities/scan-rhcos-node-host.adoc:19]: @Maddosaurus OpenShift > 4.7 right?

Comment (gaurav-nelson) [reply operating/manage-vulnerabilities/scan-rhcos-node-host.adoc:19]: @Maddosaurus we also say requires 4.10 or later at https://github.com/openshift/openshift-docs/pull/59886/files#diff-6458b7d7560d86665cede052a43c9f0e7c571f883f94049b1487410f16a3c0ccR12 Can you confirm the correct OCP version for which we support this feature.

Comment (vikin91) [reply operating/manage-vulnerabilities/scan-rhcos-node-host.adoc]: The feature works for 4.7 and later, but ACS officially supports 4.10 and later. I would suggest to write 4.10 if in doubt.

**Review by vikin91 — APPROVED** [review]
LGTM (provided we unify the supported versions to 4.10)

File: modules/rhcos-enable-node-scan.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+$ oc -n stackrox patch daemonset/collector -p '{"spec":{"template":{"spec":{"containers":[{"name":"compliance","env":[{"name":"ROX_METRICS_PORT","value":":9091"},{"name":"ROX_NODE_SCANNING_ENDPOINT","value":"127.0.0.1:8444"},{"name":"ROX_NODE_SCANNING_INTERVAL","value":"4h"},{"name":"ROX_NODE_SCANNING_INTERVAL_DEVIATION","value":"24m"},{"name":"ROX_NODE_SCANNING_MAX_INITIAL_WAIT","value":"5m"},{"name":"ROX_RHCOS_NODE_SCANNING","value":"true"},{"name":"ROX_CALL_NODE_INVENTORY_ENABLED","value":"true"}]}]}}}}'
+----
+. Run the following commands to update the Collector DaemonSet (DS):
+.. Add new volume mounts to Collector DS:
++
+[source,terminal]
+----
+$ oc -n stackrox patch daemonset/collector -p '{"spec":{"template":{"spec":{"volumes":[{"name":"tmp-volume","emptyDir":{}},{"name":"cache-volume","emptyDir":{"sizeLimit":"200Mi"}}]}}}}'
+----
+.. Add the new `NodeScanner` container:
```
Comment (lahinson) [inline modules/rhcos-enable-node-scan.adoc:35]:
```suggestion
.. Add the new `NodeScanner` container by running the following command:
```

File: operating/manage-vulnerabilities/scan-rhcos-node-host.adoc [hunk 4]
```diff
@@ -12,18 +12,20 @@ With {rh-rhacs-first}, you can scan RHCOS nodes for vulnerabilities and detect p
...
 
-First, {product-title-short} analyzes and detects RHCOS components, and then it matches vulnerabilities for identified components using Red Hat RHEL and OpenShift 4.X Open Vulnerability and Assessment Language (OVAL) v2 security data streams.
+First, {product-title-short} analyzes and detects RHCOS components, and then it matches vulnerabilities for identified components by using Red Hat RHEL and OpenShift 4.X Open Vulnerability and Assessment Language (OVAL) v2 security data streams.
 
 [NOTE]
 ====
-* Red Hat supports scanning for RHCOS nodes only if you are using it with {ocp}.
-* You must use {ocp} 4.10 or later for scanning RHCOS node hosts.
+* If you have installed {product-title-short} by using `roxctl` CLI, you must manually enable the RHCOS node scanning features. When you use Helm or Operator installation methods on {ocp}, this feature is enabled by default.
+* Red Hat supports scanning for RHCOS nodes only if you are using it with {ocp} 4.10 and later.
```
Comment (lahinson) [inline operating/manage-vulnerabilities/scan-rhcos-node-host.adoc:20]: Consider clarifying what "it" is in this sentence. For example, is this correct? "Red Hat supports scanning for RHCOS nodes only if you are using RHCOS with {ocp} 4.10 and later."

**Review by lahinson — COMMENTED** [review]
I provided a few suggestions, but overall, LGTM.

**Pull request merged**